### PR TITLE
PHP 8.0 syntax in bundles (without CoreBundle and ApiBundle)

### PR DESCRIPTION
**Pull request by Zales0123**
| Q               | A
| --------------- | -----
| Branch?         | 1.11
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | continuation of https://github.com/Sylius/Sylius/pull/13502
| License         | MIT

### CODE REVIEW

File: src/Sylius/Bundle/AddressingBundle/Validator/Constraints/ProvinceAddressConstraintValidator.php [hunk 1]
```diff
@@ -47,7 +41,7 @@ public function validate($value, Constraint $constraint): void
         $propertyPath = $this->context->getPropertyPath();
 
         foreach (iterator_to_array($this->context->getViolations()) as $violation) {
-            if (0 === strpos($violation->getPropertyPath(), $propertyPath)) {
+            if (str_starts_with($violation->getPropertyPath(), $propertyPath)) {
```
Comment (Zales0123) [inline src/Sylius/Bundle/AddressingBundle/Validator/Constraints/ProvinceAddressConstraintValidator.php:44]: OH MY GOD it feels so good ❤️

Comment (loic425) [reply src/Sylius/Bundle/AddressingBundle/Validator/Constraints/ProvinceAddressConstraintValidator.php:44]: Yes, I agree, this is a great one.

**Review by AdamKasp — APPROVED** [review]

File: src/Sylius/Bundle/AdminBundle/Controller/Dashboard/StatisticsController.php [hunk 2]
```diff
@@ -21,20 +21,11 @@
...
     /**
      * @param EngineInterface|Environment $templatingEngine
      */
-    public function __construct(
-        object $templatingEngine,
-        StatisticsDataProviderInterface $statisticsDataProvider
-    ) {
-        $this->templatingEngine = $templatingEngine;
-        $this->statisticsDataProvider = $statisticsDataProvider;
+    public function __construct(private object $templatingEngine, private StatisticsDataProviderInterface $statisticsDataProvider)
```
Comment (loic425) [inline src/Sylius/Bundle/AdminBundle/Controller/Dashboard/StatisticsController.php:27]:
```suggestion
    public function __construct(private EngineInterface|Environment $templatingEngine, private StatisticsDataProviderInterface $statisticsDataProvider)
```

But maybe this is a BC-break.

File: src/Sylius/Bundle/AdminBundle/Controller/DashboardController.php [hunk 3]
```diff
@@ -27,32 +27,16 @@
...
      * @param EngineInterface|Environment $templatingEngine
      */
     public function __construct(
-        ChannelRepositoryInterface $channelRepository,
-        object $templatingEngine,
-        RouterInterface $router,
-        ?SalesDataProviderInterface $salesDataProvider = null,
-        ?StatisticsDataProviderInterface $statisticsDataProvider = null
+        private ChannelRepositoryInterface $channelRepository,
+        private object $templatingEngine,
```
Comment (loic425) [inline src/Sylius/Bundle/AdminBundle/Controller/DashboardController.php:35]: same thing here and maybe also elsewhere

File: src/Sylius/Bundle/AdminBundle/Controller/Dashboard/StatisticsController.php (hunk 2 above)
Comment (lchrusciel) [reply src/Sylius/Bundle/AdminBundle/Controller/Dashboard/StatisticsController.php:27]: It is BC-Break :( 

On the other hand, the code won't work if anything else would be injected (unless someone duplicate the contract under different name). Also, we had "soft" requirement for these classes. I would introduce it anyway

File: src/Sylius/Bundle/AdminBundle/Controller/CustomerStatisticsController.php [hunk 4]
```diff
@@ -24,24 +24,14 @@
...
     /**
      * @param EngineInterface|Environment $templatingEngine
      */
     public function __construct(
-        CustomerStatisticsProviderInterface $statisticsProvider,
-        RepositoryInterface $customerRepository,
-        object $templatingEngine
+        private CustomerStatisticsProviderInterface $statisticsProvider,
+        private RepositoryInterface $customerRepository,
+        private object $templatingEngine
```
Comment (lchrusciel) [inline src/Sylius/Bundle/AdminBundle/Controller/CustomerStatisticsController.php:33]:
```suggestion
        private EngineInterface|Environment $templatingEngine
```

File: src/Sylius/Bundle/AdminBundle/Controller/Dashboard/StatisticsController.php (hunk 2 above)
Comment (Zales0123) [reply src/Sylius/Bundle/AdminBundle/Controller/Dashboard/StatisticsController.php]: 👍

File: src/Sylius/Bundle/AdminBundle/Controller/ImpersonateUserController.php [hunk 5]
```diff
@@ -26,32 +26,19 @@
...
-    private string $authorizationRole;
-
     public function __construct(
-        UserImpersonatorInterface $impersonator,
-        AuthorizationCheckerInterface $authorizationChecker,
-        UserProviderInterface $userProvider,
+        private UserImpersonatorInterface $impersonator,
+        private AuthorizationCheckerInterface $authorizationChecker,
+        private UserProviderInterface $userProvider,
         ?RouterInterface $router,
```
Comment (lchrusciel) [inline src/Sylius/Bundle/AdminBundle/Controller/ImpersonateUserController.php:35]:
```suggestion
        private ?RouterInterface $router,
```

Comment (lchrusciel) [reply src/Sylius/Bundle/AdminBundle/Controller/ImpersonateUserController.php:35]: But to be honest, we should rework this argument, to be able to replace logic with the new one

File: src/Sylius/Bundle/AdminBundle/Controller/ImpersonateUserController.php [hunk 6]
```diff
@@ -26,32 +26,19 @@
...
+        private string $authorizationRole
     ) {
         if (null !== $router) {
             @trigger_error('Passing RouterInterface as the fourth argument is deprecated since 1.4 and will be prohibited in 2.0', \E_USER_DEPRECATED);
         }
-
-        $this->impersonator = $impersonator;
-        $this->authorizationChecker = $authorizationChecker;
-        $this->userProvider = $userProvider;
         $this->router = $router;
```
Comment (lchrusciel) [inline src/Sylius/Bundle/AdminBundle/Controller/ImpersonateUserController.php:41]:
```suggestion
```

File: src/Sylius/Bundle/LocaleBundle/Templating/Helper/LocaleHelper.php [hunk 7]
```diff
@@ -20,25 +20,21 @@
 
 final class LocaleHelper extends Helper implements LocaleHelperInterface
 {
-    private LocaleConverterInterface $localeConverter;
-
     private ?LocaleContextInterface $localeContext;
 
-    public function __construct(LocaleConverterInterface $localeConverter, ?LocaleContextInterface $localeContext = null)
+    public function __construct(private LocaleConverterInterface $localeConverter, ?LocaleContextInterface $localeContext = null)
```
Comment (lchrusciel) [inline src/Sylius/Bundle/LocaleBundle/Templating/Helper/LocaleHelper.php:25]:
```suggestion
    public function __construct(private LocaleConverterInterface $localeConverter, private ?LocaleContextInterface $localeContext = null)
```

File: src/Sylius/Bundle/LocaleBundle/Templating/Helper/LocaleHelper.php [hunk 8]
```diff
@@ -20,25 +20,21 @@
...
     {
         if (null === $localeContext) {
             @trigger_error('Not passing LocaleContextInterface explicitly as the second argument is deprecated since 1.4 and will be prohibited in 2.0', \E_USER_DEPRECATED);
         }
-
-        $this->localeConverter = $localeConverter;
         $this->localeContext = $localeContext;
```
Comment (lchrusciel) [inline src/Sylius/Bundle/LocaleBundle/Templating/Helper/LocaleHelper.php:30]:
```suggestion
```

File: src/Sylius/Bundle/ShopBundle/Controller/SecurityWidgetController.php [hunk 9]
```diff
@@ -19,15 +19,11 @@
...
 final class SecurityWidgetController
 {
-    /** @var EngineInterface|Environment */
-    private $templatingEngine;
-
     /**
      * @param EngineInterface|Environment $templatingEngine
      */
-    public function __construct(object $templatingEngine)
+    public function __construct(private object $templatingEngine)
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ShopBundle/Controller/SecurityWidgetController.php:25]:
```suggestion
    public function __construct(private EngineInterface|Environment $templatingEngine)
```

File: src/Sylius/Bundle/ShopBundle/Controller/SecurityWidgetController.php [hunk 10]
```diff
@@ -19,15 +19,11 @@
 
 final class SecurityWidgetController
 {
-    /** @var EngineInterface|Environment */
-    private $templatingEngine;
-
     /**
      * @param EngineInterface|Environment $templatingEngine
      */
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ShopBundle/Controller/SecurityWidgetController.php:24]:
```suggestion
```

File: src/Sylius/Bundle/AdminBundle/Controller/ImpersonateUserController.php (hunk 5 above)
Comment (Zales0123) [reply src/Sylius/Bundle/AdminBundle/Controller/ImpersonateUserController.php:35]: Interesting idea, we can elaborate over it in the separate PR 🖖

**Review by lchrusciel — APPROVED** [review]

**Comment by lchrusciel** [comment]
Thank you, Mateusz! :1st_place_medal:

**Pull request merged**